File: c++/calculator/calculator.cpp
#include <iostream>

using namespace std;

double Add(double a, double b);
double Subtract(double a, double b);
double Divide(double a, double b);
double Multiply(double a, double b);

int main()
{
	int input;
	cout << "========= CALCULATOR ===========" << endl;
	cout << "1. Add" << endl;
	cout << "2. Subtract" << endl;
	cout << "3. Divide" << endl;
	cout << "4. Multiply" << endl;
	cout << "5. Exit" << endl;

	do {

		cout << "\nChoose your Arithmatic Operation : " << endl;
		cin >> input;
		int a; 
		int b;

		cout << "First Number : " << endl;
		cin >> a;

		cout << "Second Number : " << endl;
		cin >> b;

		switch(input) {
			case 1 : cout << a << "+" << b << " = " << Add(a, b) << endl; break;
			case 2 : cout << a << "-" << b << " = " << Subtract(a, b) << endl; break;
			case 3 : cout <<  a << "/" << b << " = " << Divide(a, b)  << endl; break;
			case 4 : cout <<  a << "x" << b << " = " << Multiply(a, b)  << endl; break;
			case 5 : cout << "Exit..." << endl; break;
			default : {
				cout << "Invalid input!" << endl;
			}
		}
	} while (input != 5);
	cout << "\nGoodbye..." << endl;
	return 0;
}

double Add(double a, double b) { return a + b; }
double Subtract(double a, double b) { return a - b; }
double Divide(double a, double b) { 
	if (!b) {
		return a / b;
	}
	return 0;
}

double Multiply(double a, double b) { return a * b; }

File: quickjs/calculator.js
const readline = require("readline");
const rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout
});

function Add(a, b) { return a + b; }
function Subtract(a, b) { return a - b; }
function Divide(a, b) { return b === 0 ? "Undefined" : a / b; }
function Multiply(a, b) { return a * b; }

console.log("============= CALCULATOR ===============");
console.log("1. Add");
console.log("2. Subtract");
console.log("3. Divide");
console.log("4. Multiply");
console.log("5. Exit");

function displayMenu()
{
	rl.question("\nChoose your Operation : ", (choice) => {
		if( choice === "5")
		{
			console.log("Goodbye...");
			rl.close();
			return;
		}

		rl.question("Enter first number: ", (a)=> {
			rl.question("Enter second number: ", (b) => {
				a = parseFloat(a);
				b = parseFloat(b);

				switch(choice) {
					case "1" : console.log("Result : ", Add(a, b)); break;
					case "2" : console.log("Result : ", Subtract(a, b)); break;
					case "3" : console.log("Result : ", Divide(a, b)); break;
					case "4" : console.log("Result : ", Multiply(a, b)); break;
					default: 
						console.log("Invalid input!");
				}

				displayMenu(); // loop back to menu.
			});
		});
	});
}

displayMenu();